### PR TITLE
Grog small omni fixes

### PR DESCRIPTION
**Pull request by supertick**

### CODE REVIEW

File: src/main/java/org/myrobotlab/io/FileIO.java [hunk 1]
```diff
@@ -1575,4 +1576,22 @@ public static String getExt(final String filename) {
...
+   * @param - dirPath non normalized path
+   * @return - fixed path
+   */
+  public static String normalize(String dirPath) {
+    if (dirPath == null) {
+      return null;
+    }
+    Platform platform = Platform.getLocalInstance();
+    if (platform.isWindows()) {
+      return dirPath.replace("/", "\\");
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/io/FileIO.java:1590]: should be replaceAll ... yes?

File: src/main/java/org/myrobotlab/jme3/Jme3Util.java [hunk 2]
```diff
@@ -158,48 +158,52 @@ public static Integer getIndexFromUnitVector(Vector3f vector) {
...
+      q.fromAngles(euler[0], euler[1], euler[2]);
+      n.setLocalRotation(q);
 
-    if (currentMenuView != null && n == selectedForView) {
-      currentMenuView.putText(selectedForView);
+      if (currentMenuView != null && n == selectedForView) {
+        currentMenuView.putText(selectedForView);
+      }
+    } catch (Exception e) {
+      log.error("{}.rotateTo threw", jme.getName(), e);
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/jme3/Jme3Util.java:206]: if jme is null, this log message will throw a NPE.

File: src/main/java/org/myrobotlab/process/InProcessCli.java [hunk 3]
```diff
@@ -386,15 +386,15 @@ public synchronized void stop() {
       worker.interrupt();
     }
 
-    if (in != null) {
+    if (in != null && in != System.in) {
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/process/InProcessCli.java]: I think System.in.equals(in) would be better?

File: src/main/java/org/myrobotlab/process/InProcessCli.java [hunk 4]
```diff
@@ -386,15 +386,15 @@ public synchronized void stop() {
...
       try {
         in.close();
       } catch (Exception e) {
         log.info("sdin error");
       }
     }
 
-    if (out != null) {
+    if (out != null  && out != System.out) {
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/process/InProcessCli.java]: System.out.equals(out)  would be better.

File: src/main/java/org/myrobotlab/service/InMoov2.java [hunk 5]
```diff
@@ -963,6 +918,14 @@ public void cameraOn() {
     }
   }
 
+  public void moveLeftArm(double bicep, double rotate, double shoulder, double omoplate) {
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/InMoov2.java]: should these be Double not double ?

File: src/main/java/org/myrobotlab/service/InMoov2.java [hunk 6]
```diff
@@ -963,6 +918,14 @@ public void cameraOn() {
...
+    moveArm("left", bicep, rotate, shoulder, omoplate);
+  }
+
+  public void moveRightArm(double bicep, double rotate, double shoulder, double omoplate) {
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/InMoov2.java]: Double not double ?

File: src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java [hunk 7]
```diff
@@ -595,15 +595,44 @@ public boolean moveTo(Double newPos) {
...
+
   @Override
   public Double moveToBlocking(Double pos) {
     return moveToBlocking(pos, null);
   }
 
+  @Override
+  public Double moveToBlocking(Integer newPos, Long timeoutMs) {
+    try {
+      processMove((double) newPos, true, timeoutMs);
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java:628]: Double or double ?

File: src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java [hunk 8]
```diff
@@ -1022,6 +1059,17 @@ public String publishServoEnable(String name) {
     return name;
   }
 
+  @Override
+  public boolean moveTo(Integer newPos) {
+    try {
+      moveTo((double) newPos);
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java:1065]: Double ?  newPos could be null..

File: src/main/java/org/myrobotlab/io/FileIO.java (hunk 1 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/io/FileIO.java:1590]: strangely according to the docs - replace is more appropriate

File: src/main/java/org/myrobotlab/jme3/Jme3Util.java (hunk 2 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/jme3/Jme3Util.java:206]: constructor requires jme

File: src/main/java/org/myrobotlab/process/InProcessCli.java (hunk 3 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/process/InProcessCli.java]: k

File: src/main/java/org/myrobotlab/process/InProcessCli.java (hunk 4 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/process/InProcessCli.java]: k

File: src/main/java/org/myrobotlab/service/InMoov2.java (hunk 5 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/InMoov2.java]: yes :facepalm:

File: src/main/java/org/myrobotlab/service/InMoov2.java (hunk 6 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/InMoov2.java]: yes :facepalm:

File: src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java (hunk 7 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java:628]: you can't cast Integer as a Double

File: src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java (hunk 8 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java:1065]: you can't cast Integer as a Double

**Pull request merged**